src/app_service/service/community/service.nim

```diff
@@ -240,6 +240,10 @@ QtObject:
       if (chat.categoryId == categoryId):
         let fullChatId = community.id & chat.id
         var chatDetails = self.chatService.getChatById(fullChatId)
+        # chats moved to a new category must be searched by chat.id
```

Comment:
Does it mean that the channel at the beginning is identified by "community.id & chat.id" and after moving it to the other category it is identified by "chat.id" ? As you can notice chatService contains map [chat id, chatDto]. This implies that when the chat is moved it is saved in the map under the different key, which seems to be incorrect. But correct me if I am wrong.

Reply:
Actually, channels inside a community are always `community.id & chat.id` regardless of their category.
So I'm wondering how this change helps.

Reply:
It seems that a moved channel is saved in the map under wrong key. This is my suspicion. @faust4exe I think you need to investigate how these keys are changed when the channel is moved.

Reply:
@endulab @PascalPrecht  In this particular place when moving chat to category the value of fullChatId is equal to `community.id & community.id & chat.id` as result it fails to find the chat by that wrong ID, and the chat.id itself has already the correct value of `community.id & chat.id`.

Reply:
"community.id & community.id & chat.id" seems to be incorrect. I think it's better to fix it to be always "community.id & chat.id" because we will have problems in the future.

Reply:
@endulab I mean here
```
let fullChatId = community.id & chat.id
```
you get twice the community id because chat.id already has community id in its chat.id :)
So I am not sure what other fix can be applied here